model_analyzer/analyzer.py

```diff
@@ -122,145 +126,28 @@ def run(self):
-        specifications.
-
-        Parameters
-        ----------
-        writer : OutputWriter
-            Used to write the result tables to an output stream
-        column_separator : str
-            The string that will be inserted between each column
-            of the table
-
-        Raises
-        ------
-        TritonModelAnalyzerException
-        """
-
-        for table in self._result_manager.get_all_tables().values():
-            self._write_result(table,
-                               writer,
-                               column_separator,
-                               ignore_widths=False)
-
-    def _write_result(self,
-                      table,
-                      writer,
-                      column_separator,
-                      ignore_widths=False,
-                      include_title=True):
-        """
-        Utility function that writes any table
-        """
-
-        if include_title:
-            writer.write('\n'.join([
-                table.title() + ":",
-                table.to_formatted_string(separator=column_separator,
-                                          ignore_widths=ignore_widths), "\n"
-            ]))
-        else:
-            writer.write(
-                table.to_formatted_string(separator=column_separator,
-                                          ignore_widths=ignore_widths) +
-                "\n\n")
-
-    def _export_server_only_csv(self, writer, column_separator):
-        """
-        Writes the server-only table as a csv file using the given writer
+            # Add the best measurements from the best result to the plot for the model
+            for result in self._result_manager.top_n_results(
+                    n=config.top_n_configs):
+                model_config_name = result.run_config().model_config(
+                ).get_field('name')
+                for measurement in result.top_n_measurements(
+                        n=config.top_n_measurements):
+                    self._plot_manager.add_measurement(
+                        model_config_label=model_config_name,
+                        measurement=measurement)
 
-        Parameters
-        ----------
-        writer : OutputWriter
-            Used to write the result tables to an output stream
-        column_separator : str
-            The string that will be inserted between each column
-            of the table
-
-        Raises
-        ------
-        TritonModelAnalyzerException
-        """
+            # Write plots to disk
+            self._plot_manager.complete_plots(model_name=model.model_name())
 
-        self._write_result(self._result_manager.get_server_table(),
-                           writer,
-                           column_separator,
-                           ignore_widths=True,
-                           include_title=False)
+            # Write results to tables
+            self._result_manager.compile_results()
```

Comment:
Can we rename `compile_results` to `write_results`?

Reply:
This function doesn't actual 'write results' in the way that `ResultManager.write_and_export_results` does. It just puts them into ResultTables, which was why i used the word compile. I didn't want to confuse it with writing output.